pkg/nodes/node.go

```diff
@@ -23,10 +25,75 @@ const (
 
 // Builder provides struct for Node object containing connection to the cluster and the list of Node definitions.
 type Builder struct {
-	Definition *v1.Node
-	Object     *v1.Node
-	apiClient  *clients.Settings
-	errorMsg   string
+	Definition  *v1.Node
+	Object      *v1.Node
+	apiClient   *clients.Settings
+	errorMsg    string
+	drainHelper *drain.Helper
+}
+
+// SetDrainHelper builds drain Helper that contains parameters to control the behaviour of drain.
+func (builder *Builder) SetDrainHelper(
+	force bool,
+	ignoreDaemonsets bool,
+	deleteLocalData bool,
+	gracePeriod int,
+	skipWaitForDeleteTimeoutSeconds int,
+	timeout time.Duration,
+) *Builder {
+	glog.V(100).Infof("Creating new DrainOptions config")
+
+	msg := fmt.Sprintf("Node draining configuration: 'force': %v,", force)
+	msg += fmt.Sprintf(" 'gracePeriod': %d seconds,", gracePeriod)
+	msg += fmt.Sprintf(" 'skipWaitForDeletionTimeout': %d seconds,", skipWaitForDeleteTimeoutSeconds)
+	msg += fmt.Sprintf(" 'ignoreAllDaemonSets': %v,", ignoreDaemonsets)
+	msg += fmt.Sprintf(" 'timeout': %v,", timeout)
+	msg += fmt.Sprintf(" 'deleteEmptyDir': %v", deleteLocalData)
+
+	glog.V(100).Infof(msg)
+
+	builder.drainHelper = &drain.Helper{
+		Ctx:    context.TODO(),
+		Client: builder.apiClient.K8sClient,
+		// Delete pods that do not declare a controller.
+		Force: force,
+		// GracePeriodSeconds is how long to wait for a pod to terminate.
+		GracePeriodSeconds: gracePeriod,
+		// Ignore DaemonSet-managed pods
+		IgnoreAllDaemonSets: ignoreDaemonsets,
+		// The length of time to wait before giving up
+		Timeout: timeout,
+		// Local data from emptyDir volumes will be deleted
+		// when the node is drained
+		DeleteEmptyDirData: deleteLocalData,
+		Out:                os.Stdout,
+		ErrOut:             os.Stderr,
+		// If pod DeletionTimestamp older than N seconds, skip waiting for the pod.
+		SkipWaitForDeleteTimeoutSeconds: skipWaitForDeleteTimeoutSeconds,
+	}
+
+	return builder
+}
+
+// Drain drains the node.
```

Comment:
```suggestion
// Drain evacuates pods from the node.
```

Reply:
That's not 100% accurate as evacuation of pods is one of the strategies and the other one is simply to delete pods

Reply:
evacuates or deletes (as long as it's not the same word).
I just looked for the first verb here: https://docs.openshift.com/container-platform/4.14/nodes/nodes/nodes-nodes-working.html

Reply:
does updated message work for U?